### ISSUE TITLE
The 'latest' / 'v2.0.0' Docker image doesn't contain shell

### ISSUE THREAD

**Issue by sfuerte** [hadolint/hadolint#578]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

Our Bitbucket pipeline is defined as follows:

```
    - step:
        name: Snapshot - Dockerlint
        trigger: automatic
        max-time: 5
        image: hadolint/hadolint
        script:
          - hadolint Dockerfile*
```

It was working prior to v2.0.0.

### Actual behavior

After the latest release, getting the following error at the Dockerlint step:
```
failed to create containerd task: OCI runtime create failed: container_linux.go:349: starting container process caused "exec: \"/bin/sh\": stat /bin/sh: no such file or directory": unknown
```

### Steps to reproduce the behavior

A quick test locally confirmed the absence of the default shell:

```
> docker run -v $(pwd)/Dockerfile:/Dockerfile --rm -ti hadolint/hadolint sh
docker: Error response from daemon: OCI runtime create failed: container_linux.go:370: starting container process caused: exec: "sh": executable file not found in $PATH: unknown.

> docker run -v $(pwd)/Dockerfile:/Dockerfile --rm -ti hadolint/hadolint:v2.0.0-alpine sh
/ #
```

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```bash
docker run -v $(pwd)/Dockerfile:/Dockerfile --rm -ti hadolint/hadolint hadolint -v
Haskell Dockerfile Linter v2.0.0-0-gdc61382

docker run -v $(pwd)/Dockerfile:/Dockerfile --rm -ti hadolint/hadolint:v2.0.0-alpine hadolint -v
Haskell Dockerfile Linter v2.0.0-0-gdc61382
```

While I can fix the issue with locking to a particular version but then we'd need to monitor your version change, keep track of it in our repos and update every single time. Definitely time consuming and inefficient. 

Is it possible to revert the change and embed a shell in the default - `latest` - Docker image? 
Or at least create a tag like `latest-alpine` which would contain the shell and be version agnostic.

Thanks!

**Comment by eberg23** [comment]
+1

**Comment by eberg23** [comment]
Workaround: split the command line args into different args. In my case, this was changing...
```
  commands:
  - hadolint --config hadolint.yaml drone-plugin/Dockerfile
```
to...
```
  command:
  - hadolint 
  - --config 
  - hadolint.yaml
  - drone-plugin/Dockerfile
  - ```

**Comment by m-ildefons** [comment]
> Workaround: split the command line args into different args in a `command:` block. In my case, this was changing...

As far as I can tell, that won't work for @sfuerte , because `hadolint Dockerfile*` is relying on the shells globbing to expand.
This whole situation was never supposed to work and only did work, because the hadolint:v1.xx-alpine image was wrongly tagged as the hadolint:v1.xx image as well and that has been fixed in the last release.

There are hadolint images with moving tags for the alpine and debian distributions on ghcr.io:
`ghcr.io/hadolint/hadolint:latest-alpine`
We might be able to push those to dockerhub for future releases as well.

**Comment by sfuerte** [comment]
it's not about shell globbing and expansion, it's also that Bitbucket pipelines are hard-coded to execute a Shell first and treat each line under `script` as a separate command.
https://support.atlassian.com/bitbucket-cloud/docs/configure-bitbucket-pipelinesyml/

E.g. the following will be executed one by one and not merged together:
```
    - step:
        name: Snapshot - eslint, test & dependency checks
        trigger: automatic
        max-time: 5
        image: node:alpine
        caches:
          - node
        script:
          - export CI=true
          - node -v
          - npm -v
          - npm install
          - npm run build
          - npm run lint
          - npm test
```

But will check `ghcr.io` images, thanks for the hint!

**Comment by lorenzo** [comment]
Please use the Debian or alpine based image. Unfortunately adding a shell to that image was an accident that happened for a few versions and was recently corrected.

**Comment by sfuerte** [comment]
@lorenzo, I'm fine with using Debian or Alpine based image as long as there is `latest-alpine`, i.e. version agnostic and no need to update the pipeline definitions upon each release.

`ghcr.io` works like a charm though, I guess we can switch to those. But still, would be great to see the same tags on Dockerhub.

```
    - step:
        name: Snapshot - Dockerlint
        trigger: automatic
        max-time: 5
        image: ghcr.io/hadolint/hadolint:latest-alpine
        script:
          - hadolint Dockerfile*
```

![image](https://user-images.githubusercontent.com/16941995/112385721-91526f00-8cb5-11eb-8b78-2dc8844a146c.png)

**Comment by m-ildefons** [comment]
There is one. Its just that its a bit hard to find between the other versions:
```
$ docker run --rm -it hadolint/hadolint:latest-alpine sh
/ # hadolint --version
Haskell Dockerfile Linter v2.0.0-0-gdc61382
```

**Comment by sfuerte** [comment]
Fair enough. I guess we can close it then.
Thanks!
